lib/Models/prev_workout_data.dart

```diff
@@ -64,34 +68,39 @@ class PrevWorkoutData {
 
   List<PrevExerciseCard> setExerciseWidgets() {
     exerciseWidgets = [];
-    for (Exercise exercise in workout.exercises) {
-      List<PrevSetWidget> setList = [];
-      int i = 0;
-      for (Sets _ in exercise.sets) {
-        setList.add(new PrevSetWidget(
-            name: exercise.name,
-            exercise: exercise,
-            addNewSet: addSet,
-            id: i,
-            isTemplate: workout.template));
-        i++;
+    if (workout.exercises is List<Exercise>) {
```

Comment:
why is this check here? Shouldn't workout.exercises always be a list of Exercises? If not we should probably just change it so it definitely is instead of adding a check every time we want to use it

Reply:
@jorre127 because in some strange cases there are listo of Strings - id of exercise not proper Exercise instance, maybe we should look at this closer

Reply:
Ok i removed those ifs, because problem was in loading workoutdata, and if you try to load exercise with real exerciseId but Firebase returns null there was a problem. 

I fix this with removing failure exerciseId from list, and problem with strings in workout.exercises in further code disappear :P